charger/ocpp.go

```diff
@@ -375,21 +389,34 @@ func (c *OCPP) updatePeriod(current float64, phases int) error {
 
 	current = math.Trunc(10*current) / 10
 
-	err = c.setChargingProfile(c.connector, getTxChargingProfile(current, phases, txn))
+	err = c.setChargingProfile(c.connector, c.getTxChargingProfile(current, txn))
 	if err != nil {
 		err = fmt.Errorf("set charging profile: %w", err)
 	}
 
 	return err
 }
 
-func getTxChargingProfile(current float64, phases, transactionId int) *types.ChargingProfile {
-	period := types.NewChargingSchedulePeriod(0, current)
+func (c *OCPP) getTxChargingProfile(current float64, transactionId int) *types.ChargingProfile {
+	var phases = c.phases
+	var value float64
+	if c.chargingRateUnit == types.ChargingRateUnitWatts {
+		voltage := 230.0
+		// Get number of currently active phases from Loadpoint
+		phases = c.lp.GetPhases()
+		if phases == 0 {
+			phases = 3
+		}
+		value = math.Trunc(current * float64(phases) * voltage)
+	} else {
+		// if phases == 0 {
+		// 	phases = 3
+		// }
+		value = current
+	}
 
-	// TODO add phases support
-	// if phases != 0 {
-	// 	period.NumberPhases = &phases
-	// }
+	period := types.NewChargingSchedulePeriod(0, value)
+	// period.NumberPhases = &phases
```

Comment:
Das wäre falsch falls der Ladepunkt mitteilt, dass ein 2p Fahrzeug angeschlossen ist und hier in W gesteuert würde?

Reply:
Deswegen mein Jein. Die Anzahl der Phasen ist eigentlich immer wichtig, egal, welche Einheit zum Steuern benutzt wird. Wenn ich die Phasen nur mitgebe, wenn in W gesteuert wird, würde eine geeignete Box auch eine Phasenumschaltung machen, das könnte ich dann nicht verhindern.
Wie willst Du es jetzt haben?

Reply:
Lass due Phasen bitte weg- anderer PR. Unabhängig davon wäre der kommentierte Code im Beispiel falsch und gehört nicht in diesen PR.

Reply:
Aber ist ok. Ich verstehe was Du mit der Änderung auf c.phases sagen willst. Nur korrekt ist es nicht ;)

Reply:
Ich kann Dir nicht folgen. Die Phasen sind doch weg. Hast Du ggf. mal ein paar Minuten für einen Chat in Slack?